R/prerun.R

```diff
@@ -109,3 +109,91 @@ hash_info_inputs <- function(inputs) {
   }, USE.NAMES = FALSE)
   digest::digest(inputs$md5sum[order(inputs$role)])
 }
+
+prerun <- function(queue) {
+  function(input) {
+    files <- jsonlite::fromJSON(input, simplifyVector = FALSE)
+    all_files <- c(files$inputs, files$outputs)
+    lapply(c(names(all_files)), function(name) {
+      file <- all_files[[name]]
+      if (!file.exists(file$path)) {
+        hintr_error(sprintf(paste0(
+          "File '%s' at path '%s' with original name '%s' ",
+          "does not exist. Make sure to upload it first ",
+          "with '/internal/upload/*' endpoints."),
+          name, file$path, file$filename),
+          "PRERUN_MISSING_FILES")
+      }
+    })
+
+    model_fit_output <- naomi:::build_hintr_output(
+      NULL,
+      files$outputs$fit_model_output$path,
+      warnings = NULL)
+    calibrate_output <- naomi:::build_hintr_output(
+      files$outputs$calibrate_plot_data$path,
+      files$outputs$calibrate_model_output$path,
+      warnings = NULL)
+    prerun_build_state(queue, files$inputs, model_fit_output, calibrate_output)
+  }
+}
+
+prerun_build_state <- function(queue, inputs, model_fit_output,
+                               calibrate_output) {
+  output <- naomi::read_hintr_output(calibrate_output$model_output_path)
+  model_fit_options <- yaml::read_yaml(text = output$info$options.yml)
+  calibration_options <- yaml::read_yaml(text =
+                                           output$info$calibration_options.yml)
+  packages <- utils::read.csv(text = output$info$packages.csv)
+
+  inputs <- build_state_inputs(inputs)
```

Comment:
not really following the "build" naming here?

Reply:
Constructing the state json, from inputs, model fit, calibrate, version etc.